py/persistentcode.c

```diff
@@ -471,7 +475,7 @@ void mp_raw_code_load(mp_reader_t *reader, mp_compiled_module_t *cm) {
         || header[3] > MP_SMALL_INT_BITS) {
         mp_raise_ValueError(MP_ERROR_TEXT("incompatible .mpy file"));
     }
-    if (MPY_FEATURE_DECODE_ARCH(header[2]) != MP_NATIVE_ARCH_NONE) {
+    if (arch != MP_NATIVE_ARCH_NONE) {
```

Comment:
It looks like the check now ignores the MSB of `arch`.  It would be good to reserve that bit for future use, and so force it to be 0 here.

An easy way to do that would be:
```c
#define MPY_FEATURE_DECODE_ARCH(feat) (((feat) >> 2) & 0x2F)
```